scipy/stats/tests/test_multivariate.py

```diff
@@ -3751,6 +3751,21 @@ def test_dimensionality_one(self):
         assert_equal(dist.var(), 0)
         assert_equal(dist.cov(), 0)
 
+    def test_n_is_zero(self):
+        # similarly, only one possible outcome if n is zero
+        n = 0
+        alpha = np.asarray([1., 1.])
+        x = np.asarray([0, 0])
+        dist = dirichlet_multinomial(alpha, n)
+
+        assert_equal(dist.pmf(x), 1)
+        assert_equal(dist.pmf(x+1), 0)
+        assert_equal(dist.logpmf(x), 0)
+        assert_equal(dist.logpmf(x+1), -np.inf)
+        assert_equal(dist.mean(), [0, 0])
```

Comment:
For reference, `tensorflow-probability` seems to allow an `n` of `0` and produce the same mean, but `pymc` still documents `n > 0` and complains (though doesn't error out) when using `n == 0`:

```python
import tensorflow_probability as tfp
import pymc


alpha = [1., 1.]
n = 0
dist = tfp.distributions.DirichletMultinomial(n, alpha)
res = dist.mean()
print("res:", res)

with pymc.Model() as model:
    X = pymc.DirichletMultinomial("X", n, alpha)

with model:
    trace = pymc.sample(draws=100, tune=10)
```

```
<snip>
res: tf.Tensor([0. 0.], shape=(2,), dtype=float32)
Multiprocess sampling (4 chains in 4 jobs)
Metropolis: [X]
Sampling 4 chains, 0 divergences ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00 / 0:00:00
Sampling 4 chains for 10 tune and 100 draw iterations (40 + 400 draws total) took 0 seconds.
/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/arviz/stats/diagnostics.py:596: RuntimeWarning: invalid value encountered in scalar divide
  (between_chain_variance / within_chain_variance + num_samples - 1) / (num_samples)
```